Automattic/jetpack
# Feature: Podcast Player

Remaining work on Podcast Player Block

#### Changes proposed in this Pull Request:
* Renamed "episode" to "track" for consistency (#15211) 
* Forces refetch of feed when identical URL is submitted (#15213)
*  Use audio element from me.js rather than building our own (#15215)
* Scoped podcast player styles for consistency (#15201)
* Made feed fetch call cancellable and cancel on Block removal (#15228)
* Prevents accidental triggering of audio when selecting the block (#15226)
* Updated podcast header rendering of server-side  fallback (#15221)
* Improved accessibility and Interactions (#15207)
* Added block keywords 
* Now truncates podcast description if too long (#15253)
* Updated error notice text when user is trying to embed a Spotify podcast (#15254)
* Documented template variables (#15250)
* Updated podcast colors client side (#15249)
* Added loading and default states (#15234)

#### Is this a new feature or does it add/remove features to an existing part of Jetpack?
* Updates Podcast Player block to be ready for `experimental` status upgrade

#### Testing instructions:

_**NOTE: it should be tested in Simple sites**_

General URL for testing: https://anchor.fm/s/9400d7c/podcast/rss

- Submit the same feed URL after failed submit attempt, make sure it fetches again.
  _Tip: Can be tested via blocking request or going offline for the first request._
- Confirm there are no errors when the block is removed before the request has finished.
  _Tip: In the Network Tools switch to "Slow 3G mode" to make it easier to test._ 
- In the editor, make sure that the first click only selects the block rather than e.g. starting playback when the play button or track element is clicked.
- On the front-end, make sure the block renders correctly with JavaScript disabled:
  - Cover art, first episode title, podcast title, and description should be rendered within Player's header,
  - The audio element should not be rendered,
  - Episodes list should be a simple list of links,
  - Every color and other custom block settings should be respected.
- Make sure the Player's accessibility and interactions are correctly announced:
  - Clicking a track should announce `Loading: [Track title] [Track Description]`.
  - Current track (currently selected) in most screen readers will announce `[track title], currently track`. In Safari with VoiceOver, it announces `[track title], current item`.
  - Clicking a currently playing track should announce `Paused`.
  - The error should announce `Error: Episode unavailable. (Open in a new window)`.
    _Tip: You can simulate an error by going offline and trying to play an episode._
- Make sure track description doesn't exceed 4 lines and is truncated with an ellipsis.
  _Tip: Use this URL https://investlikethebest.libsyn.com/rss (it has looooong descriptions)._
- Make sure that, when trying to embed a Spotify podcast, correct error notice is shown:
  > It looks like you're trying to embed a podcast hosted on Spotify. Please use the Spotify block instead.

  _Tip: Here's a Spotify podcast URL for you https://open.spotify.com/show/4XPl3uEEL9hvqMkoZrzbx5?si=8noFQsHjTpCMjtA-djqmMg._
- Make sure the custom colors are being set in both editor and front-end.
- Make sure the loading state is applied correctly:
  - Test on the frontend with Podcast Player block present
  - Disable JavaScript in your browser and reload
  - You should see very basic podcast player interface sans any interactive controls (as described above)
  - To check the loading state, in dev tools, remove class is-default from the block
- Make sure that all variables used in files in templates directory are properly documented in their headers.
#### Proposed changelog entry for your changes:
* Not sure if needed at this time?

#### Follow-up issues
- #15271 Break components into their own files
- #15272 Fix React memo optimizations
- #15275 Improve non-compatible URL detection
